sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -271,36 +260,36 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
       // Fix misplacement when comment is at the end of the query.
       .map(_.split("\n").filterNot(_.startsWith("--")).mkString("\n")).map(_.trim).filter(_ != "")
 
-    // When we are regenerating the golden files for test cases without '--IMPORT' specified, or
-    // running test cases against [[ThriftServerQueryTestSuite], we don't need to set any config as
-    // they all need to return the same result.
-    // When we use '--SET' and '--IMPORT' together for those import queries, we want to run the
-    // same queries from the original file but with different settings and save the answers. So the
-    // `--SET` will be respected in this case.
-    if ((regenerateGoldenFiles && importedTestCaseName.isEmpty) || !isTestWithConfigSets) {
-      runQueries(queries, testCase, None)
+    val settingLines = comments.filter(_.startsWith("--SET ")).map(_.substring(6))
+    val settings = settingLines.flatMap(_.split(",").map { kv =>
+      val (conf, value) = kv.span(_ != '=')
+      conf.trim -> value.substring(1).trim
+    })
+
+    if (regenerateGoldenFiles) {
```

Comment:
We can remove this line:
https://github.com/apache/spark/blob/3be4734f317b3b26da278ad088b31ec4b7750213/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L114